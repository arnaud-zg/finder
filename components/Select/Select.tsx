import { rem } from 'csx'
import { Field } from 'formik'
import React from 'react'
import { style } from 'typestyle'
import { colors } from '../../constants/colors'

const className = style({
  backgroundColor: colors.light.toHexString(),
  border: 0,
  borderColor: colors.gray.toHexString(),
  borderRadius: rem(0.25),
  borderStyle: 'solid',
  borderWidth: rem(0.0625),
  color: colors.gray.toHexString(),
  fontSize: rem(1),
  marginLeft: rem(0.25),
  marginTop: rem(0.5),
  padding: rem(0.25),
})

interface IProps extends React.HTMLProps<HTMLSelectElement> {
  options: JSX.Element[]
}

export const Select = (props: IProps) => {
  const { options } = props
  return (
    <Field component="select" name={props.name} className={className}>
      {options}
    </Field>
  )
}
