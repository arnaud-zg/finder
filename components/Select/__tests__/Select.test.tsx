import { Formik } from 'formik'
import React from 'react'
import renderer from 'react-test-renderer'
import { Select } from '../Select'

describe('TextInput', () => {
  it('renders correctly', () => {
    const testRenderer = renderer.create(
      <Formik
        initialValues={{ mode: '' }}
        onSubmit={() => {}}
        render={() => (
          <Select
            name="mode"
            options={[
              <option value="Level 1" />,
              <option value="Level 2" />,
              <option value="Level 3" />,
              <option value="Boss" />,
            ]}
          />
        )}
      />
    )
    expect(testRenderer.toJSON()).toMatchSnapshot()
  })
})
