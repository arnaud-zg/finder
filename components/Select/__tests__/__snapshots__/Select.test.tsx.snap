// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput renders correctly 1`] = `
<select
  className="foacnk6"
  name="mode"
  onBlur={[Function]}
  onChange={[Function]}
  value=""
>
  <option
    value="Level 1"
  />
  <option
    value="Level 2"
  />
  <option
    value="Level 3"
  />
  <option
    value="Boss"
  />
</select>
`;
